index.js

```diff
@@ -382,6 +384,25 @@ class Hyperbee {
     return b.del(key, opts)
   }
 
+  watch (range, onchange) {
+    if (typeof range === 'function') {
+      onchange = range
+      range = undefined
+    }
+
+    const watcher = new Watcher(this, range)
+
+    this.feed.once('close', () => {
```

Comment:
Not sure how these event handlers work, but what happens to this event if the watcher closes, but the feed does not? Will the event still exist? If so, that's a (tiny) memory leak

Reply:
Sounds like it is, I will add a test creating hundreds of watchers